File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.java
class Solution {
    public int diagonalSum(int[][] mat) {
        int sum = 0;
        int len = mat[0].length;
        
        for(int i = 0; i < len; i++)
            sum += mat[i][len-1-i] + mat[i][i];
        if(len % 2 == 1)
            sum -= mat[len/2][len/2];
        return sum;
    }
}

File: 1572-matrix-diagonal-sum/1572-matrix-diagonal-sum.swift
class Solution {
    func diagonalSum(_ mat: [[Int]]) -> Int {
        var res = 0
        var i = 0
        var j = mat[0].count-1
        
        while i < mat.count {
            if i != j {
                res += (mat[i][i] + mat[i][j])
            } else {
                res += mat[i][i]
            }
            i += 1
            j -= 1
        }
        return res
    }
}